.github/workflows/check-pr.yml

```diff
@@ -0,0 +1,30 @@
+name: Check PR
+
+on: pull_request
+permissions: {}
+
+concurrency:
+  group: check-pr-${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  unit-tests:
+    name: Run tests
+    uses: ./.github/workflows/unit-tests.yml
+
+  sonarcloud:
+    name: Scan
+    needs: unit-tests
+    uses: ./.github/workflows/sonarcloud.yml
+    secrets:
+      github-token: ${{ secrets.GITHUB_TOKEN }}
+      sonar-token: ${{ secrets.SONAR_TOKEN }}
+
+  pre-commit:
+    name: pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run pre-commit
```

Comment:
What was the reasoning for creating our own GHA, as opposed to using the existing & functional but deprecated one [supplied by pre-commit](https://github.com/pre-commit/action)?

Reply:
The mains reasons are that it's deprecated and using our own also allows us to reduce duplication since we can re-use one shared action in all the repos.